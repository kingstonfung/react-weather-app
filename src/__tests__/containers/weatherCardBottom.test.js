import React from 'react';
import ReactDOM from 'react-dom';

import WeatherCardBottom from '../../containers/WeatherCardBottom';

const mockForecastData = [{
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}, {
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}, {
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}, {
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}, {
  "temp": {
    "min": -5.95,
    "max": -1.64,
  },
  "weather": [{
    "main": "Snow",
  }],
}];

describe('WeatherCard component...', () => {
  const staticWeatherCardBottom = <WeatherCardBottom forecast={mockForecastData}/>;

  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(staticWeatherCardBottom, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
});