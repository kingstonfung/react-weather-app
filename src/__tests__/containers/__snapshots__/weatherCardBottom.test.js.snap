// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard component... should match snapshot 1`] = `
<div>
  <div
    class="cardBottomPortion"
  >
    <div
      class="cardDetail"
    >
      <ul
        class="forecastsWrapper"
      >
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Thu
          </p>
        </li>
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Fri
          </p>
        </li>
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Sat
          </p>
        </li>
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Sun
          </p>
        </li>
        <li
          class="forecastItem"
        >
          <span
            class="forecastIcon wi wi-wu-snow"
          />
          <div
            class="cardTemperature"
          >
            -2
          </div>
          <div
            class="cardTemperature"
          >
            -6
          </div>
          <p
            class="forecastDay"
          >
            Mon
          </p>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
